addons/missionTesting/functions/fnc_runWeaponsTest.sqf

```diff
@@ -0,0 +1,158 @@
+                if ((_x select 0) == _magazineClassname) then {
+                    _hasMags = true;
+                    _usedAmmo pushBack _magazineClassname;
+                    _count = _count + (_x select 1);
+                };
+            } forEach (magazinesAmmoFull _unit);
+            if (_count > 0) then {
+                private _magPic = getText (configFile >> "CfgMagazines" >> _magazineClassname >> "picture");
+                _muzzleText = _muzzleText + format ["<img image='%1' width='16' height='16'/>:%2 ", _magPic, _count];
+            };
+        } forEach _mags;
+        if (_weaponClassname == "throw") then {
+            if (_hasMags) then {
+                _unitText = _unitText + _muzzleText;
+            };
+        } else {
+            _unitText = _unitText + _muzzleText + "]<br/>";
+        };
+    } forEach _muzzles;
+    if (_weaponClassname == "throw") then {
+        _unitText = _unitText + "]<br/>";
+    };
+};
+
+{
+    private _unit = _x;
+    private _classname = typeOf _unit;
+
+    if (!(_classname in _handledClasses)) then {
+        _handledClasses pushBack _classname;
+
+        private _displayName = getText (configfile >> "CfgVehicles" >> _classname >> "displayName");
+        private _xIcon = gettext (configfile >> "CfgVehicles" >> _classname >> "icon");
+        private _image = gettext (configfile >> "CfgVehicleIcons" >> _xIcon);
+        private _color = "";
+        private _sideArray = [];
+
+        switch (side _x) do {
+        case (west): {
+                _color = "#0088EE"; //0,0.45,0.9,1
+            };
+        case (east): {
+                _color = "#DD0000"; //0,0.45,0.9,1
+            };
+        case (resistance): {
+                _color = "#00DD00"; //0,0.45,0.9,1
+            };
+            default {
+                _color = "#880099"; //0,0.45,0.9,1
+            };
+        };
+        private _unitText = format ["<t color='%1' size='1'>%2</t> - %3<br/>", _color, _displayName, _classname];
+        private _usedAmmo = [];
+        [primaryWeapon _unit] call _fncGetWeaponInfo;
+        [secondaryWeapon _unit] call _fncGetWeaponInfo;
+        [handgunWeapon _unit] call _fncGetWeaponInfo;
+        ["throw"] call _fncGetWeaponInfo;
+
+        private _extraAmmo = (magazines _unit) - _usedAmmo;
+        if ((count _extraAmmo) > 0) then {
+            _unitText = _unitText + "[Extra: ";
+            {
+                private _magPic = getText (configFile >> "CfgMagazines" >> _x >> "picture");
+                _unitText = _unitText + format ["<img image='%1' width='16' height='16'/> ", _magPic];
+            } forEach _extraAmmo;
+            _unitText = _unitText + "]<br/>";
+        };
+        private _has343 = [_unit, "ACRE_PRC343"] call acre_api_fnc_hasKindOfRadio;
+        private _has148 = [_unit, "ACRE_PRC148"] call acre_api_fnc_hasKindOfRadio;
+        _unitText = _unitText + format ["[Radios: %1%2]<BR/>", ["", "343 "] select _has343, ["", "148"] select _has148];
```

Comment:
No check for 117?

Reply:
Again copied form Pabo's weapons test. Can be easily added if it is required.

Reply:
I'd recommend it. Loadouts can easily squeeze out a 117 and I've had many times in the past forgotten about it until testing with other people